src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java

```diff
@@ -215,4 +220,54 @@ private static Patient extractPatient(ResultSet rs) throws SQLException
 			throw new SQLException(e);
 		}
     }
+
+    public static void removePatientsEverywhereInStudy(int internalStudyId, Set<String> patientStableIds) throws DaoException {
+        log.info("Removing {} patients from study with internal id={} ...", patientStableIds, internalStudyId);
+        Set<Integer> internalPatientIds = findInternalPatientIdsInStudy(internalStudyId, patientStableIds);
+        for (Integer internalPatientId : internalPatientIds) {
+            Set<Integer> patientInternalSampleIds = DaoSample.getSamplesByPatientId(internalPatientId).stream()
+                    .map(Sample::getInternalPatientId).collect(Collectors.toSet());
+            DaoGeneticAlteration.getInstance().removeSamplesInGeneticAlterationsForStudy(internalStudyId, patientInternalSampleIds);
```

Comment:
I think `removeSamplesInGeneticAlterationsForStudy` should be logic belonging to deleteSamples and deletePatients should just call deleteSamples.

Reply:
Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/bd54ed240186925755cfcaa16e50ee4a255a1d21